contracts/mesh-consumer/src/ibc.rs

```diff
@@ -173,25 +173,53 @@ pub fn receive_unstake(
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn ibc_packet_ack(
     deps: DepsMut,
-    _env: Env,
+    env: Env,
     msg: IbcPacketAckMsg,
 ) -> Result<IbcBasicResponse, ContractError> {
-    // we only care if there was an error...
     let res: StdAck = from_slice(&msg.acknowledgement.data)?;
-    if res.is_ok() {
-        return Ok(IbcBasicResponse::new());
+    if res.is_err() {
+        return Err(ContractError::AckFailed {});
     }
 
-    // we need to parse the ack based on our request
+    // We need to parse the ack based on our request
     let original_packet: ConsumerMsg = from_slice(&msg.original_packet.data)?;
     match original_packet {
-        ConsumerMsg::Rewards {} => fail_rewards(deps),
-        ConsumerMsg::UpdateValidators { added, removed } => {
-            fail_update_validators(deps, added, removed)
-        }
+        ConsumerMsg::Rewards {
+            validator: _,
+            total_funds,
+        } => acknowledge_rewards(deps, env, total_funds),
+        ConsumerMsg::UpdateValidators {
+            added: _,
+            removed: _,
+        } => Ok(IbcBasicResponse::new()),
     }
 }
 
+pub fn acknowledge_rewards(
+    deps: DepsMut,
+    env: Env,
+    amount: Coin,
+) -> Result<IbcBasicResponse, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let timeout: IbcTimeout = env.block.time.plus_seconds(300).into();
+
+    // NOTE We try to split the addr from the port_id, maybe better to set the addr in init?
```

Comment:
I was actually thinking about it some days ago (not related to mesh-security), I was thinking about passing a struct as a version and getting some data out of it, can be very useful.
Should be possible because you do the verification, and if its not, it should be built to be possible lol